src/i2c.rs

```diff
@@ -568,6 +609,118 @@ pub enum I2cSingleMasterErr {
     // Other errors like the 'clock low timeout' UCCLTOIFG may appear here in future.
 }
 
+/// List of possible I2C interrupt sources. 
+/// 
+/// Used when reading from the I2C interrupt vector register via [`interrupt_source()`](I2cSingleMaster::interrupt_source())
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub enum I2cVector {
+    /// No interrupt.
+    None                = 0x00,
+    /// Arbitration was lost during an attempted transmission.
+    ArbitrationLost     = 0x02,
+    /// Received a NACK.
+    NackReceived        = 0x04,
+    /// Received a Start condition on the I2C bus along with one of our own addresses.
+    StartReceived       = 0x06,
+    /// Received a Stop condition on the I2C bus.
+    /// This is usually set when acting as an I2C slave, but that this can occur as an I2C master during a zero byte write.
+    StopReceived        = 0x08,
+    /// Slave address 3 received a data byte.
+    Slave3RxBufFull     = 0x0A,
+    /// The Tx buffer is empty and slave address 3 was on the I2C bus when this occurred.
+    Slave3TxBufEmpty    = 0x0C,
+    /// Slave address 2 received a data byte.
+    Slave2RxBufFull     = 0x0E,
+    /// The Tx buffer is empty and slave address 2 was on the I2C bus when this occurred.
+    Slave2TxBufEmpty    = 0x10,
+    /// Slave address 1 received a data byte.
+    Slave1RxBufFull     = 0x12,
+    /// The Tx buffer is empty and slave address 1 was on the I2C bus when this occurred.
+    Slave1TxBufEmpty    = 0x14,
+    /// Data is waiting in the Rx buffer. In slave mode slave address 0 was on the I2C bus when this occurred.
+    RxBufFull           = 0x16,
+    /// The Tx buffer is empty. In slave mode slave address 0 was on the I2C bus when this occurred.
+    TxBufEmpty          = 0x18,
+    /// The target byte count has been reached.
+    ByteCounterZero     = 0x1A,
+    /// The SCL line has been held low longer than the Clock Low Timeout value.
+    ClockLowTimeout     = 0x1C,
+    /// The 9th bit of an I2C data packet has been completed.
+    NinthBitReceived    = 0x1E,
+}
+
+/// Human-friendly list of possible I2C interrupt source flags.
+/// 
+/// Used for writing to the I2C interrupt enable register e.g. via the [`set_interrupts()`](I2cSingleMaster::set_interrupts()) method.
+/// 
+/// Example usage:
+/// ```ignore
+/// use I2cInterruptBits::*;
+/// i2c_device.set_interrupts(RxBufFull | TxBufEmpty | StopReceived);
+/// ```
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(u16)]
+pub enum I2cInterruptBits {
```

Comment:
Should we just import `bitflags` and get rid of most of this boilerplate? We're basically replicating this functionality.